### PR TITLE
Fix mech requests criteria for rewards eligibility

### PR DESCRIPTION
**Pull request by truemiller**
there are no contract errors thrown with the updated changes @oaksprout unsure if this is enough to merge to main, but the formula makes sense, and has been approved thus far. 

image

### CODE REVIEW

File: frontend/service/Autonolas.ts [hunk 1]
```diff
@@ -134,16 +140,22 @@ const getAgentStakingRewardsInfo = async ({
 const getAvailableRewardsForEpoch = async (): Promise<number | undefined> => {
   const contractCalls = [
     serviceStakingTokenMechUsageContract.rewardsPerSecond(),
-    serviceStakingTokenMechUsageContract.livenessPeriod(),
+    serviceStakingTokenMechUsageContract.livenessPeriod(), // epoch length?
+    serviceStakingTokenMechUsageContract.tsCheckpoint(), // last checkpoint timestamp?
```
Comment (truemiller) [inline frontend/service/Autonolas.ts:144]: i'm assuming the last checkpoint timestamp is roughly the same as the start of the current epoch?

Comment (jmoreira-valory) [reply frontend/service/Autonolas.ts:144]: livenessPeriod is a constant for each staking program. It's the minimum epoch lenght (minimum call between checkpoints)

tsCheckpoint is the latest executed checkpoint - start of a new epoch.

**Review by jmoreira-valory — APPROVED** [review]

File: frontend/service/Autonolas.ts [hunk 2]
```diff
@@ -134,16 +140,22 @@ const getAgentStakingRewardsInfo = async ({
 const getAvailableRewardsForEpoch = async (): Promise<number | undefined> => {
   const contractCalls = [
     serviceStakingTokenMechUsageContract.rewardsPerSecond(),
-    serviceStakingTokenMechUsageContract.livenessPeriod(),
+    serviceStakingTokenMechUsageContract.livenessPeriod(), // epoch length?
```
Comment (truemiller) [inline frontend/service/Autonolas.ts:143]:
```suggestion
    serviceStakingTokenMechUsageContract.livenessPeriod(), // epoch length
```

File: frontend/service/Autonolas.ts (hunk 1 above)
Comment (truemiller) [inline frontend/service/Autonolas.ts:144]:
```suggestion
    serviceStakingTokenMechUsageContract.tsCheckpoint(), // last checkpoint timestamp
```

**Review by oaksprout — APPROVED** [review]

**Pull request merged**